client/my-sites/domains/components/domain-form-fieldsets/region-address-fieldsets.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External	 dependencies
+ *
+ * @format
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import identity from 'lodash/identity';
+import noop from 'lodash/noop';
+
+/**
+ * Internal dependencies
+ */
+import {
+	CHECKOUT_EU_ADDRESS_FORMAT_COUNTRY_CODES,
+	CHECKOUT_UK_ADDRESS_FORMAT_COUNTRY_CODES,
+} from './constants';
+import UsAddressFields from './us-address-fieldset';
+import EuAddressFields from './eu-address-fieldset';
+import UkAddressFields from './uk-address-fieldset';
+
+const regionAddressFieldsetComponents = [
+	{
+		keys: CHECKOUT_EU_ADDRESS_FORMAT_COUNTRY_CODES,
+		component: EuAddressFields,
+	},
+	{
+		keys: CHECKOUT_UK_ADDRESS_FORMAT_COUNTRY_CODES,
+		component: UkAddressFields,
+	},
+];
+
+/**
+ * regionAddressFieldsetComponent Objects
+ * @typedef {Object} regionAddressFieldsetDefinitionObject
+ * @property {Array<String>} keys array of country codes
+ * @property {ReactElement} component corresponding address for the country code
+ */
+
+/**
+ * @param {Array<regionAddressFieldsetDefinitionObject>} formTypesArray array of region address fieldset components and keys
+ * @param {String} key countryCode
+ * @returns {ReactElement} the address field component for the countryCode
+ */
+function getAddressComponent( formTypesArray, key ) {
+	const formTypesArrayLength = formTypesArray.length;
+	let Component;
+	for ( let i = 0; i < formTypesArrayLength; i++ ) {
+		if ( formTypesArray[ i ].keys.indexOf( key ) > -1 ) {
+			Component = formTypesArray[ i ].component;
+			break;
+		}
+	}
+	return Component;
+}
+
+const RegionAddressFieldsets = ( { getFieldProps, translate, countryCode } ) => {
+	const Component =
+		getAddressComponent( regionAddressFieldsetComponents, countryCode ) || UsAddressFields;
+	return (
+		<Component
```

Comment:
While I actually wasn't aware that basically using a variable her is possible, I'd discourage you doing so, as it makes it hard to see what's going on. We don't have a React component `<Component>`, yet you are successfully using one.

I assume this is to avoid having three very similar (i.e. same variables) HTML fields, yet I encourage you to still us a construct like
```
if ( countryCode in CHECKOUT_EU_ADDRESS_FORMAT_COUNTRY_CODES ) {
	return ( <EuAddressFields
		getFieldProps={ getFieldProps }
		translate={ translate }
		countryCode={ countryCode }
	>);
}
if ( countryCode in CHECKOUT_UK_ADDRESS_FORMAT_COUNTRY_CODES ) {
	return ( <UkAddressFields
		getFieldProps={ getFieldProps }
		translate={ translate }
		countryCode={ countryCode }
	>);
}
```

Reply:
No probs. All for making things more readable! :) Cheers.